internal/controller/compaction/status_compaction.go

```diff
@@ -0,0 +1,97 @@
+package compaction
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/core/v1alpha1"
+	druidmetrics "github.com/gardener/etcd-druid/internal/metrics"
+
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func (r *Reconciler) triggerFullSnapshotAndUpdateStatus(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd, accumulatedEtcdRevisions, triggerFullSnapshotThreshold int64) (ctrl.Result, error) {
+	latestSnapshotCompactionCondition := druidv1alpha1.Condition{
```

Comment:
You can use a simpler variable name like `condition` since that's the only condition updated by this controller, and makes the code easier to read.